File: src/test/java/com/cp/csms/authentication/validation/TokenExistenceValidatorTest.java
package com.cp.csms.authentication.validation;

import com.cp.csms.common.AuthenticationMessage;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.InjectMocks;
import org.mockito.junit.jupiter.MockitoExtension;

import static org.assertj.core.api.Assertions.assertThat;

@ExtendWith(MockitoExtension.class)
public class TokenExistenceValidatorTest {

    private static final String REQUEST_ID = "req-123";

    @InjectMocks
    private TokenExistenceValidator validator;

    @Test
    void shouldReturnTrueWhenTokenExists() {
        final AuthenticationMessage message = new AuthenticationMessage(REQUEST_ID, "valid-token");

        final boolean result = validator.validate(message);

        assertThat(result).isTrue();
    }

    @Test
    void shouldReturnFalseWhenMessageIsNull() {
        final boolean result = validator.validate(null);

        assertThat(result).isFalse();
    }

    @Test
    void shouldReturnFalseWhenTokenIsNull() {
        final AuthenticationMessage message = new AuthenticationMessage(REQUEST_ID, null);

        final boolean result = validator.validate(message);

        assertThat(result).isFalse();
    }

    @Test
    void shouldReturnFalseWhenTokenIsEmpty() {
        final AuthenticationMessage message = new AuthenticationMessage(REQUEST_ID, "");

        final boolean result = validator.validate(message);

        assertThat(result).isFalse();
    }

    @Test
    void shouldReturnFalseWhenTokenIsBlank() {
        final AuthenticationMessage message = new AuthenticationMessage(REQUEST_ID, "   ");

        final boolean result = validator.validate(message);

        assertThat(result).isFalse();
    }

    @Test
    void shouldReturnFalseWhenTokenIsOnlyWhitespace() {
        final AuthenticationMessage message = new AuthenticationMessage(REQUEST_ID, "\t\n\r");

        final boolean result = validator.validate(message);

        assertThat(result).isFalse();
    }
}

File: src/test/kotlin/com/cp/csms/authentication/validation/TokenExistenceValidatorTest.kt
package com.cp.csms.authentication.validation

import com.cp.csms.common.AuthenticationMessage
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.mockito.InjectMocks
import org.mockito.junit.jupiter.MockitoExtension

@ExtendWith(MockitoExtension::class)
class TokenExistenceValidatorTest {

    @InjectMocks
    private lateinit var validator: TokenExistenceValidator

    @Test
    fun `should return true when token exists`() {
        val message = AuthenticationMessage(REQUEST_ID, "valid-token")

        val result = validator.validate(message)

        assertThat(result).isTrue()
    }

    @Test
    fun `should return false when message is null`() {
        val result = validator.validate(null)

        assertThat(result).isFalse()
    }

    @Test
    fun `should return false when token is null`() {
        val message = AuthenticationMessage(REQUEST_ID, null)

        val result = validator.validate(message)

        assertThat(result).isFalse()
    }

    @Test
    fun `should return false when token is empty`() {
        val message = AuthenticationMessage(REQUEST_ID, "")

        val result = validator.validate(message)

        assertThat(result).isFalse()
    }

    @Test
    fun `should return false when token is blank`() {
        val message = AuthenticationMessage(REQUEST_ID, "   ")

        val result = validator.validate(message)

        assertThat(result).isFalse()
    }

    @Test
    fun `should return true when token has leading and trailing spaces but is not blank`() {
        val message = AuthenticationMessage(REQUEST_ID, "  valid-token  ")

        val result = validator.validate(message)

        assertThat(result).isTrue()
    }

    companion object {
        private const val REQUEST_ID = "req-123"
    }
}
